### PR TITLE
feat: disable "Add Project" when judging starts

### PR DESCRIPTION
**Pull request by jkaster**
- Judging time for the hackathon is always shown at the bottom of the project list now
- If judging has started:
  - the judging start time is displayed to the left of the `Add Project` button
  - if the user has the ability to lock projects, `Add Project` remains enabled
  - if the user is a regular hackathon attendee, `Add Project` is disabled

- Made intro agenda on the home page display English or Japanese depending on the user's locale

# Screenshots

## Regular hackathon attendee

image

## User who can lock projects

image

### CODE REVIEW

**Review by jeremytchang — APPROVED** [review]
LGTM. We still have to lock projects when judging time comes. Ideally this should all be consolidated into one automatic flow, where when judging time starts, projects are auto locked and projects additions is stopped for hackers. These flows would not apply for staff and admin.

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx [hunk 1]
```diff
@@ -63,6 +67,23 @@ export const ProjectsScene: FC<ProjectSceneProps> = () => {
     if (hackathon) dispatch(lockProjects(false, hackathon._id))
   }
 
+  useEffect(() => {
```
Comment (jeremytchang) [inline packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:70]: Not sure if useEffect is needed here. Unless we are triggering project scene re-render periodically, this is not needed.

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx [hunk 2]
```diff
@@ -63,6 +67,23 @@ export const ProjectsScene: FC<ProjectSceneProps> = () => {
...
+    if (hackathon && hacker) {
+      const started = hackathon.judging_starts?.getTime() < new Date().getTime()
+      setJudgingStarts(
+        `Judging start${started ? 'ed' : 's'} ${zonedLocaleDate(
```
Comment (jeremytchang) [inline packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:74]: `started ? 'ed' : 's'` is dead code given conditional rendering at line 97. 
With that said, I say we remove the conditional rendering at line 97 and always show when project judging will begin.

Also need to add `at` so it reads `Judging start(s)(ed) at date).

**Review by jeremytchang — CHANGES_REQUESTED** [review]
LGTM with changes.

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx (hunk 2 above)
Comment (jkaster) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:74]: I am undecided where judging start time should show in the final implementation, so I'm only showing it for now when it has an impact on the project board. If I added "at" to be grammatically correct, I'd also need to add "on".

So I'm going with a colon instead.

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx (hunk 1 above)
Comment (jkaster) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:70]: I may not understand `useSelector()` well enough yet, but it seems `hackathon` and `hacker` may not be established when this page is first opened. Is that correct?

Comment (jeremytchang) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:70]: Redux state change will automatically kick off a render. 
`When an action is dispatched, useSelector() will do a reference comparison of the previous selector result value and the current result value. If they are different, the component will be forced to re-render. If they are the same, the component will not re-render.`
https://react-redux.js.org/api/hooks
There's more nuances when dealing with state comparision, but that shouldn't matter here, you can test it to double check.

Comment (jeremytchang) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:70]: Also afaict, project scene is not rendered or can be navigated to until hacker/hackathon is all loaded in the redux state in the first place. Hence the long lag between app opening and project tab appearing.
(Something to fix later)

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx [hunk 3]
```diff
@@ -73,7 +94,16 @@ export const ProjectsScene: FC<ProjectSceneProps> = () => {
...
+          </Span>
+        )}
+        <Button
+          iconBefore={<Add />}
+          onClick={handleAdd}
+          disabled={(isLoading || judgingStarted) && !canLockProject(hacker)}
+        >
           Add Project
         </Button>
         <>
```
Comment (jeremytchang) [inline packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:109]: The line below this one already demonstrates dependence on hackathon and hacker props to render. Also do not be confused by useSelector. It is almost the same as mapState, which maps state to props on the component. If a prop changes, the component re-renders.

**Review by jeremytchang — CHANGES_REQUESTED** [review]
I wrote up an example of what the code could be. 

Also moved the text to after the button. From previous frontend work, typically statuses appear after the button, so the button is not shifted when status changes (though that doesn't apply in this case, since the button will never shift). It also reads better, from left to right. You see the Add button disabled, then you read the status or reason for it.

https://github.com/looker-open-source/sdk-codegen/blob/jc/add_project_example/packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx (hunk 3 above)
Comment (jkaster) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx:109]: Thanks for the explanation about `useSelector()`. I think there may be other places with `useEffect()` that don't need it, but we can worry about that when we switch to the new persistence layer. It is possible someone could be on the project page when they reload, and not move from it until after it successfully loads, but we have a "not found" scenario that already handles that. Less code is more.

**Review by jeremytchang — APPROVED** [review]
LGTM

File: packages/hackathon/src/scenes/HomeScene/HomeScene.tsx [hunk 4]
```diff
@@ -41,6 +42,16 @@ const MARKDOWN_LINEBREAK = '  '
 
 export const HomeScene: FC<HomeSceneProps> = ({ hacker }) => {
   const schedule = localAgenda(hacker.locale)
+  const host = getExtensionSDK().lookerHostData?.hostUrl
+  const intro =
+    hacker.locale === 'ja_JP'
+      ? `### ハッカソン詳細については、[よくある質問記事](https://community.looker.com/hackathome-2021-1026/hackathome-2021-%E3%82%88%E3%81%8F%E3%81%82%E3%82%8B%E8%B3%AA%E5%95%8F-28518)をご確認いただけます。
```
Comment (jeremytchang) [inline packages/hackathon/src/scenes/HomeScene/HomeScene.tsx:48]: Where did these translation come from?

**Review by jeremytchang — APPROVED** [review]
LGTM

File: packages/hackathon/src/scenes/HomeScene/HomeScene.tsx (hunk 4 above)
Comment (jkaster) [reply packages/hackathon/src/scenes/HomeScene/HomeScene.tsx:48]: From a PR @lanlooker did last night. 🪄

File: packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx (hunk 1 above)
Comment (jkaster) [reply packages/hackathon/src/scenes/ProjectsScene/ProjectsScene.tsx]: Good stuff. Thanks for the cleaner version. I put it into this PR.

**Pull request merged**